Task3/Task3_Nikolay_Varadinov/src/python/car_wash/lib_py_pool/py_pool.py

```diff
@@ -0,0 +1,151 @@
+"""Module with Read and Write Connection Pool."""
+
+import logging
+import time
+import typing
+
+from contextlib import contextmanager
+from sys import exc_info
+from threading import Lock
+
+import psycopg2
+
+from . meta_classes import SingletonMeta
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class PoolConn(metaclass=SingletonMeta):
+
+    """Pool which is python list and strores DB connections."""
+
+    @staticmethod
+    def stringer(db_config: dict) -> str:
+        """Create string of key word arguments from dict."""
+
+        config = (" ").join(
+            f"{key}={value}" for key, value in db_config.items())
+        return config
+
+    def __init__(self, size: int, *args, **kwargs) -> None:
+        """Accept db_config as dict in args or as kwargs."""
+
+        self.db_config = (args, kwargs)
+        self.pool = []
+        self.size = size
+        self._lock = Lock()
+        self._populate_pool()
+
+    @property
+    def db_config(self):
+        """Nicer __init__ with this."""
+
+        return self._db_config
+
+    @db_config.setter
+    def db_config(self, config_tuple: tuple[tuple[dict], dict]):
+        """Checks if db_config is given as dict or kwargs."""
+
+        if len(config_tuple[0]) != 0:
+            self._db_config = self.stringer(config_tuple[0][0])
+        else:
+            self._db_config = self.stringer(config_tuple[1])
+
+    def _create_connection(self) -> psycopg2.connect:  # type: ignore
+        """Create database connection with psycopg2."""
+
+        return psycopg2.connect(self.db_config)
+
+    def _pull(self) -> psycopg2.connect:  # type: ignore
+        """Pull connection. Using Lock to prevent the race conditions."""
+
+        with self._lock:
+            while len(self.pool) == 0:
```

Comment:
It means all processes will wait 1 second.....
https://github.com/mehalyna/CarWashing-System/pull/38/files#r928255340